array_api_strict/tests/test_array_object.py

```diff
@@ -101,6 +101,30 @@ def test_validate_index():
     assert_raises(IndexError, lambda: a[idx])
 
 
+@pytest.mark.parametrize("device", [None, "CPU_DEVICE", "device1", "device2"])
+@pytest.mark.parametrize(
+    "integer_index",
+    [
+        1,
+        np.bool(1),
+        np.int8(0),
+        np.uint8(0),
+        np.int16(0),
+        np.uint16(0),
+        np.int32(0),
+        np.uint32(0),
+        np.int64(0),
+        np.uint64(0),
```

Comment:
Possibly overdid it, this may not be necessary ?
Also is there any integer index types i missed?

Reply:
nb removed `np.bool(1)` (which has a `__index__` method) as this gives warning:
"FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar."